src/memo/memo.mli

```diff
@@ -29,19 +44,59 @@ end
     are forgotten, pending computations are cancelled. *)
 val reset : unit -> unit
 
-module type S = Memo_intf.S with type stack_frame := Stack_frame.t
 module type Input = Memo_intf.Input
-module type Output = Memo_intf.Output
-module type Decoder = Memo_intf.Decoder
+module type Data = Memo_intf.Data
 
-module Make(Input : Input)(Decoder : Decoder with type t := Input.t)
-  : S with type input := Input.t
+module Function : sig
```

Comment:
I don't believe this type is used outside this module. Perhaps we shouldn't include it in the mli?

Reply:
Removed.